src/core/PluginManager.cpp

```diff
@@ -136,26 +138,35 @@ namespace collabvm::core {
 					case PluginLoadError::NotServerPlugin: {
 						// Try loading as a coreplugin next
 						return HandlePluginLoad<true, false>(so);
-					}; break;
-
+						break;
+					}
+					
 					case PluginLoadError::NotCorePlugin: {
 						// Try loading as a controllerplugin.
 						return HandlePluginLoad<false, true>(so);
-					}; break;
+						break;
+					}
 				}
 
 				return res;
 			}
+			// FIXME: This is a fallback return to prevent compile time warnings from occurring.
+			return res;
```

Comment:
would a `default: break;` not be sufficient in this case? just my thought

Reply:
There is some sort of possibility which causes gcc to print a "no return on non-void function" warning during compile time. I think that gcc is finding a possibility where the if statement never occurs.

Reply:
clang-equiv: `warning: non-void function does not return a value in all control paths [-Wreturn-type]`

Interesting.